src/Connection/Http2Connection.php

```diff
@@ -309,6 +310,12 @@ private function request(Request $request, CancellationToken $token): Promise
         return call(function () use ($id, $request, $token): \Generator {
             $this->pendingRequests[$id] = $deferred = new Deferred;
 
+            if ($this->socket->isClosed()) {
+                throw new SocketException('Socket closed before request was made');
+            }
+
+            $this->socket->reference();
+
```

Comment:
I've applied this to `master`, because a test was failing on Travis with this error.

https://github.com/amphp/http-client/commit/31904f8f83f1cd8d9bf75adb17ea6ea538880b1e